flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -229,14 +232,33 @@ public GetShardListResult getShardList(Map<String, String> streamNamesWithLastSe
 	 * {@inheritDoc}
 	 */
 	@Override
-	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, @Nullable String startingSeqNum) throws InterruptedException {
+	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, @Nullable Object startingMarker) throws InterruptedException {
+		GetShardIteratorRequest getShardIteratorRequest = new GetShardIteratorRequest()
+			.withStreamName(shard.getStreamName())
+			.withShardId(shard.getShard().getShardId())
+			.withShardIteratorType(shardIteratorType);
+
+		switch (ShardIteratorType.fromValue(shardIteratorType)) {
+			case TRIM_HORIZON:
+			case LATEST:
+				break;
+			case AT_TIMESTAMP:
+				getShardIteratorRequest.setTimestamp((Date) startingMarker);
```

Comment:
For the new implementation of this method, we probably should handle type casting exceptions, and wrap them as `IllegalArgumentException`s. Since we're doing the property config validating locally, this really shouldn't happen, but it would be good to add handling type cast errors here to make this class self-contained.